### ISSUE TITLE
none-editable markdown cell can be 'unrendered'

### ISSUE THREAD

**Issue by lexnederbragt** [jupyter/notebook#2389]
When setting a markdown cell's metadata to 'editable:false', clicking in the cell 'unrenders' the html to markdown. Re-rendering needs to be done by executing the cell. On the one hand, intuitively, an uneditable markdown cell should not have this behaviour. On the other hand, being able to copy the markdown content may be desirable.

Is the current behaviour intentional or a bug? If intentional, would it be possible to have a metadata setting disabling seeing markdown for an uneditable markdown cell?

**Comment by gnestor** [comment]
So the issue is that double-clicking an uneditable markdown cell renders it as markdown but doesn't allow the user to render it back as HTML? If so, that doesn't sound right. I would expect to be able to _execute_ it back to HTML. However, I would expect the textarea to have a readonly attribute so that users cannot edit the content.

**Comment by gnestor** [comment]
I just tested myself and it appears to work as _I would expect_:

![](http://g.recordit.co/a5lvxENWST.gif)

**Comment by lexnederbragt** [comment]
Yes, that describes my experience. If that is the expected/desired behaviour, then my issue is that I'd like the possibility to prevent rendering cells with metadata 'editable:false' to be rendered as markdown. Maybe a setting 'htmlonly:true'? Could be useful for any type of cell?

My usecase is when I want to have students learning programming work from a notebook mixing text and both executable and unexecutable code, with cells they can do assignments in/try out things in. Especially real beginners get confused when they accidentally click in a rendered markdown cell and see it go back to pure markdown.

**Comment by takluyver** [comment]
Tricky one. I think I would also expect that non-editable markdown cells are fixed in the rendered view, but I can also see that the unrendered view is useful if you want to copy the raw markdown source. Maybe we should provide a different way to show the source in that case that's less easy to trigger accidentally.

**Comment by grovduck** [comment]
This doesn't address the issue within Jupyter notebook itself, but I'd thought I'd add that the `freeze` extension within `jupyter_contrib_nbextensions` works really well for this ([link](https://github.com/ipython-contrib/jupyter_contrib_nbextensions/tree/master/src/jupyter_contrib_nbextensions/nbextensions/freeze)).  My use case sounds very similar to that of @lexnederbragt.  Specifically, it adds these metadata fields to frozen markdown cells:

```
   "cell_type": "markdown",
   "metadata": {
    "deletable": false,
    "editable": false,
    "run_control": {
     "frozen": true
    }
   },
```
Maybe not necessary to address if it's handled within an extension?
